app/templates/partials/applications_table.html

```diff
@@ -2,59 +2,114 @@
 {%- from "govuk_frontend_jinja/components/tag/macro.html" import govukTag -%}
 {%- from "govuk_frontend_jinja/components/table/macro.html" import govukTable -%}
 
-{% macro applications_table(applications, is_past_submission_deadline) %}
+{% macro applications_table(applications, is_past_submission_deadline, showLanguageColumn) %}
 
-{% set ns = namespace(rows = []) %}
-    {% for application in applications %}
-    {% set status_class %}
-    {% if is_past_submission_deadline %}
-        {% if application.status == 'SUBMITTED' %}
-            complete-tag
-        {% elif application.status == 'NOT_SUBMITTED' %}
-            complete-tag
-        {% endif %}
-    {% else %}
-        {%  if application.status == 'SUBMITTED' %}
-            complete-tag
-        {%  elif application.status == 'NOT_STARTED' %}
-            not-started-tag
-        {%  elif application.status == 'IN_PROGRESS' %}
-            in-progress-tag-new
-        {%  elif application.status == 'READY_TO_SUBMIT' %}
-            in-progress-tag-new
-        {% endif %}
-    {% endif %}
-{% endset %}
+	{% set ns = namespace(rows = []) %}
+	{% for application in applications %}
+		{% set status_class %}
+		{% if is_past_submission_deadline %}
+			{% if application.status == 'SUBMITTED' %}
+				complete-tag
+			{% elif application.status == 'NOT_SUBMITTED' %}
+				complete-tag
+			{% endif %}
+		{% else %}
+			{%  if application.status == 'SUBMITTED' %}
+				complete-tag
+			{%  elif application.status == 'NOT_STARTED' %}
+				not-started-tag
+			{%  elif application.status == 'IN_PROGRESS' %}
+				in-progress-tag-new
+			{%  elif application.status == 'READY_TO_SUBMIT' %}
+				in-progress-tag-new
+			{% endif %}
+		{% endif %}
+		{% endset %}
 
-{% if not application.status == 'SUBMITTED' and not is_past_submission_deadline %}
-    {% set application_link = '<a class="govuk-link" href="{}">{}</a>'.format(url_for('application_routes.tasklist', application_id=application.id), gettext('Continue application')) %}
-{% endif %}
+		{% if not application.status == 'SUBMITTED' and not is_past_submission_deadline %}
+			{% set application_link = '<a class="govuk-link" href="{}">{}</a>'.format(url_for('application_routes.tasklist', application_id=application.id), gettext('Continue application')) %}
+		{% endif %}
 
-    {% set row = ns.rows.append(
-    [
-        {
-            'text': application.project_name if application.project_name else gettext('Untitled project')
-        },
-        {
-            'text':  govukTag({
-                'text': application.status|snake_case_to_human ,
-                'classes': status_class
-            })
-        },
-        {
-            'html': application.last_edited|datetime_format_short_month if application.last_edited != None
-        },
-        {
-            'html': application.reference
-        },
-        {
-            'html': application_link
-        }
-    ])%}
-    {% endfor %}
+		{% if showLanguageColumn %}
```

Comment:
This popped into my head after work so commenting it before I forget on Monday lol.

A way to do optional dict inside array based on a condition, to remove the big if blocks:

```python
def test(condition):
    return [
        {a:1},   
        *([{b:2}] if condition else []),
        {c:3},
    ]

test(True)   # [{a:1}, {b:2}, {c:3}]
test(False)  # [{a:1}, {c:3}]

```

Reply:
Unfortunately i havent been able to get this to work in nunjuck (not sure if it does). Have left it as it is for now